tldextract/tldextract.py

```diff
@@ -187,8 +187,15 @@ def __call__(self, url):
             .rstrip(".")
 
         is_punycode = netloc.startswith('xn--') or '.xn--' in netloc
+        # Some IDN urls might generate an error
         if is_punycode:
-            netloc = codecs.decode(netloc.encode('ascii'), 'idna')
+            try:
+                netloc = codecs.decode(netloc.encode('ascii'), 'idna')
+            except UnicodeError:
+                LOG.warn('UnicodeError when decoding netloc ' + netloc)
```

Comment:
Yes, I agree with you as well. 

I also support the RTFM / RTFS and force users pass the expected args. Unfortunately, I also don't like when my python scripts crash with no apparent reason and when I stumbled upon those weird domains, it made tldextract crash. 

Problem is a bit more "complex" because a domain that starts with xn-<blabla> can be either a punycode or just .. .a normal domain! Punycode domains are"valid" as long as users want them to be "valid". From this lib's point of view, we don't care if a punycode subdomain is valid or not, we just need to split them. But here comes the problem... in order to try to match a punycode domain from PSL we need to... convert it if possible. At least we need to try. And if we fail.. then bad luck, it wasn't punycode :) But our job is still to split that URL.Hence the need to try/catch conversion. 

Just wanted to clarify what is my reasoning in case there will be future problems :)